core/src/main/java/com/datastax/oss/driver/api/core/tracker/RequestTracker.java

```diff
@@ -48,4 +48,32 @@ void onError(
       long latencyNanos,
       DriverConfigProfile configProfile,
       Node node);
+
+  /**
+   * Invoked each time a request fails at the node level. Similar to onError but at a per node
+   * level.
+   *
+   * @param latencyNanos the overall execution time (from the {@link Session#execute(Request,
+   *     GenericType) session.execute} call until the error is propagated to the client).
```

Comment:
I completely missed this on my first review: the latency passed to node-level methods should be measured from the time we sent the request to that node, not from the beginning of the whole request cycle. Right now it includes all tries on previous nodes.

There is already a `NodeResponseCallback.start` field for metrics, you can just use that instead of `startTimeNanos` (the name `start` is not great by the way, maybe we could rename it to `nodeStartTimeNanos` for clarity).

That also means we won't be able to handle the final node in `setFinalError`, so instead just add a call to `trackNodeError` everywhere `setFinalError` is called in the `NodeResponseCallback`.